.github/workflows/staging_deploy.yml

```
@@ -4,6 +4,12 @@ on:
   push:
     branches:
       - master
+  workflow_dispatch:
+    inputs:
+      branch_or_tag:
```

```suggestion
      commit:
```

maybe something like this, since we can use anything that references a commit?